src/simpleMask.js

```diff
@@ -0,0 +1,171 @@
+const simpleMask = (containerEl, options = {}) => {
+    const defaults = {
+        format: null,
+        inputSelector: null,
+        maskPlaceholder: 'X',
+        numeric: false,
+        alphanumeric: false,
+        customPattern: null,
+        alphanumericPattern: /^[a-zA-Z0-9\.]*$/,
+        numberPattern: /[0-9\/]+/,
+        onMaskCallback: null,
+        onFailedInputCallback: null,
+    };
+
+    const settings = { ...defaults, ...options };
+
+    const maskMethods = {
+        reverse: 'reverse',
+        forward: 'forward'
+    }
+
+    const availableCharacterPatterns =  [settings.numeric, settings.alphanumeric, !!settings.customPattern]
+    const requiredParams = [containerEl, settings.inputSelector, settings.format]
+
+
+    let inputEl;
+    let output;
+    let builtValue;
+    let formatArrayOfCharacters;
+    let arrayOfSpecialCharactersInFormat;
+
+
+    const init = () => {
+        if (settingsAreValid()) {
+            inputEl = containerEl.querySelector(settings.inputSelector);
+            initPatternData();
+            bindEvents();
+        } else {
+            throw new Error('Missing one or many of the following: containerEl, input selector, numeric, alphanumeric or custom pattern')
+        }
+
+    };
+
+    const settingsAreValid = () => {
+        const selectedCharacterPattern = availableCharacterPatterns.filter(pattern => pattern === true);
```

Comment:
[suggestion] consider an [array.every()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every) here, then you don't have to worry about comparing the array length on line 50.

The other benefit (which you don't really need here, but is good for other situations), is that I _believe_ and array.every will bail if one of the items doesn't pass the condition. This is useful if you have a super big list, you get a tiny performance improvement. Again, not as important here 😄

Reply:
@scheul93 The reason I am doing it this way is to make sure that only one character pattern is passed. The length comparison also takes care of the scenario when no character patterns are passed.  So I don't think **array.every** helps me in this situation, unless I'm missing something.

Reply:
[question] Do you need three different character patterns? If only one character pattern is allowed, why not just have a single `pattern` setting?